pkg/provider/plugin/proto/plugin.proto

```diff
@@ -0,0 +1,152 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package plugin;
+option go_package = "github.com/external-secrets/external-secrets/pkg/provider/plugin/proto";
+
+// SecretsPlugin defines the gRPC interface for external secret plugins.
+service SecretsPluginService {
+  // GetInfo returns plugin information and capabilities.
+  rpc GetInfo(GetInfoRequest) returns (GetInfoResponse);
+
+  // GetSecret retrieves a single secret from the plugin.
+  rpc GetSecret(GetSecretRequest) returns (GetSecretResponse);
```

Comment:
Forgot to mention: You're modelling the API around the current interface. Could you please amend it and follow the V2 interfaces? 

https://github.com/external-secrets/external-secrets/pull/4792/files#diff-aa596699ff60cfb41536e327095186d562382b78f099618e726dcd550e618e8fR223-R238

This is still in draft but i think they'll more or less stay like that.